src/ado2gh/Program.cs

```diff
@@ -23,15 +25,41 @@ public static async Task Main(string[] args)
                 .AddSingleton<AdoApiFactory>()
                 .AddSingleton<GithubApiFactory>()
                 .AddSingleton<RetryPolicy>()
+                .AddSingleton<VersionChecker>()
+                .AddSingleton<IVersionProvider, VersionChecker>()
                 .AddHttpClient();
 
             var serviceProvider = serviceCollection.BuildServiceProvider();
-
             var parser = BuildParser(serviceProvider);
 
+            try
+            {
+                await LatestVersionCheck(serviceProvider);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogWarning($"Could not retrieve latest ado2gh version from github.com, please ensure you are using the latest version by downloading it from https://github.com/github/gh-gei/releases/latest");
+                Logger.LogVerbose(ex.ToString());
+            }
+
             await parser.InvokeAsync(args);
         }
 
+        private static async Task LatestVersionCheck(ServiceProvider sp)
+        {
+            var versionChecker = sp.GetService<VersionChecker>();
```

Comment:
```suggestion
            var versionChecker = sp.GetRequiredService<VersionChecker>();
```
this way it will throw an exception if it wasn't able to resolve the `VersionChecker` for any reason.